src/latent_gp.jl

```diff
@@ -0,0 +1,30 @@
+"""
+    LatentGP(f<:GP, x<:AbstractVector, Φ)
+
+ - `f` is a Gaussian Process.
+ - `x` is an `AbstractVector` of inputs of type `Tx`.
+ - `Φ` is the log likelihood function which maps input of type `Tx` to `Real`.
+    
+"""
+struct LatentGP{T<:GP,Tx}
+    f::T
+    x::AbstractVector{Tx}
+    σ²
+    Φ
+    function LatentGP(f::GP, x::AbstractVector{Tx}, σ², Φ) where Tx
+        return new{typeof(f),Tx}(f, x, σ², Φ)
+    end
+end
+
+function LatentGP(f::GP, x::AbstractVector{Tx}, Φ) where Tx
+        return LatentGP(f, x, 0.1, Φ)
+    end
+
+
+function mean(fx::LatentGP)
+    return AbstractGPs.mean(fx.f, fx.x)
+end
+
+function cov(fx::LatentGP)
+    return AbstractGPs.cov(fx.f, fx.x)
+end
```

Comment:
I don't think that it makes sense to provide these functions in this package. I think we need to discuss this a bit further in #1 to make sure that we're on the same page here.

Reply:
If we were to maintain `Distributions.jl` like API, we should probably keep these?

Reply:
Ideally we would do this, but it's not generally going to be clear how to compute either the mean or covariance over the output (in the proposal 2 sense). For example if the likelihood is some horrible non-linear thing.